src/detect-dataset.h

```diff
@@ -36,8 +36,15 @@
 typedef struct DetectDatasetData_ {
     Dataset *set;
     uint8_t cmd;
+    int thread_ctx_id;
 } DetectDatasetData;
 
+typedef struct DetectDatasetMatchData_ {
+    const uint8_t *data;
```

Comment:
I wonder if we should just have a pointer to `InspectionBuffer` here, or even just an index into the array of buffers

Reply:
DetectEngineContentInspectionInternal uses `const uint8_t *buffer, const uint32_t buffer_len` and not an `InspectionBuffer`